### ISSUE TITLE
Invalid MAP-T source address generated from rules

### ISSUE THREAD

**Issue by dangeist** [ayourtch/nat46#37]
We've been using the module with success for a while with (example) MAP rules similar to the following:

config map0 
    local.v4 x2.212.252.0/24 
    local.v6 xx00:8809:a504::/46 
    local.style MAP 
    local.ea-len 14 local.psid-offset 4 
    remote.v4 0.0.0.0/0 
    remote.v6 xx00:8809:bfff:ffff::/64 
    remote.style RFC6052 
    remote.ea-len 0 remote.psid-offset 0 debug 0 
    local.pd xx00:8809:a504:10::/60 
    local.psid 1

We recently switches the CE to a smaller environment which (mathematically) should work:
config map0     
    local.v4 x2.212.255.0/26 
    local.v6 xx00:8809:a50c::/48 
    local.style MAP  
    local.ea-len 12     
    local.psid-offset 4 
    remote.v4 0.0.0.0/0 
    remote.v6 xx00:8809:bfff:fffe::/64 
    remote.style RFC6052 
    remote.ea-len 0 remote.psid-offset 0 debug 0 
    local.pd xx00:8809:a50c:10::/60 
    local.psid 1

But our calculated MAP-T source address on translated packets is getting a value of:
    xx00:8809:a50c:0100:0000:48d4:ff00:0010

As this is outside the assigned PD for the CE, traffic is getting sent out and never returning (to that device).

We're looking at the **xlate_map_v4_to_v6**() **bitarray_copy**() functions to see if there's a logic error somewhere but it's a bit to get through if anyone has any thoughts as to what's going on.

**Comment by dangeist** [comment]
This ended up being the code providing input to the module not calculating the PSID correctly. Once that was addressed, things started working as expected.

**Comment by KeenoHao** [comment]
Hello, I would like to ask, where can I configure the configuration file to make nat46 work properly? Do you need to create a separate file? Or use the echo config < ifname > < cfg-strings > command to add to < cfg-strings >?

**Comment by ayourtch** [comment]
Yes you need to put the correct info into the corresponding proc file, have a look at the readme in the module subdirectory. Closing this since it was supposed to be for a different issue and I forgot to close it :)
